File: python/unittest2/modules/testfunctions.py
#!/usr/bin/env python

# FUNCTION DEFINITIONS

def hello_world():
    '''Returns "Hello, world!"'''
    return 'Hello, world!'

def reverse(mystring):
    '''Returns a reversed copy of the string passed in.'''
    mylist = list(mystring)
    mylist.reverse()
    return ''.join(mylist)

# UNIT TESTS
import unittest

class TestFunctions(unittest.TestCase):

    def test_hello_world(self):
        self.assertEqual('Hello, world!', hello_world())

    def test_reverse(self):
        self.assertEqual('54321', reverse('12345'))
        self.assertEqual('edcba', reverse('abcde'))
        self.assertRaises(TypeError, reverse(["a", "b", "c"]))
        #self.assertRaises(TypeError, reverse(12345)) # wrong error type?

if __name__ == '__main__':
    unittest.main()

File: ruby/sort_string/test_functions.rb
#!/usr/bin/env ruby

require 'test/unit'
require 'functions'

# Unit tests for the methods in module +functions+.
class TestFunctions < Test::Unit::TestCase
    # Unit tests for the +sort_string+ function.
    def test_sort_string
        assert_equal '1 22 333', sort_string('1 22 333')
        assert_equal '1 22 333', sort_string('333 22 1')
        assert_equal 'o tw thr', sort_string('o tw thr')
        assert_equal 'o tw thr', sort_string('thr tw o')
    end

    # Unit tests for the +sort_string_2+ function.
    def test_sort_string_2
        assert_equal '1 22 333', sort_string_2('1 22 333')
        assert_equal '1 22 333', sort_string_2('333 22 1')
        assert_equal 'o tw thr', sort_string_2('o tw thr')
        assert_equal 'o tw thr', sort_string_2('thr tw o')
    end
end
